### PR TITLE
ev3dev: add sound event support

### PR DESCRIPTION
**Pull request by kortschak**
@dlech Please take a look.

This is very pared down, for example only `SND_TONE` is supported, since only it seems to work on the ev3. I have not tested this on evb.

### CODE REVIEW

File: ev3dev.go [hunk 1]
```diff
@@ -71,6 +71,9 @@ const (
 
 	// PowerSupplyPath is the path to the ev3 power supply file system.
 	PowerSupplyPath = "/sys/class/power_supply"
+
+	// SoundPath is the path to the ev3 sound events.
+	SoundPath = "/dev/input/by-path/platform-snd-legoev3-event"
```
Comment (dlech) [inline ev3dev.go]: This path is EV3-specific. There will be a different path for EVB.

File: speaker.go [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+	ev, err := os.Open(SoundPath)
+	if err != nil {
+		return false, fmt.Errorf("ev3dev: failed to open sound event device: %v", err)
+	}
+	defer ev.Close()
+
+	var buf [evBufLen]byte
+	err = ioctl(ev.Fd(), eviocgbit(0, ev_max), reflect.ValueOf(buf[:]).Index(0).Addr().Pointer())
+	if err != nil {
+		return false, fmt.Errorf("ev3dev: failed to set ioctl command for sound event device: %v", err)
```
Comment (dlech) [inline speaker.go]: This error message doesn't really make sense to me. I would expect it to say something like "eviocgbit ioctl for sound event device failed"

File: speaker.go [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+	ev_snd = 0x12
+	ev_max = 0x1f
+
+	evBufLen = (ev_max + 7) / 8
+)
+
+// Tone plays a tone at the specified frequency from the ev3 speaker.
+// If freq is zero, playing is stopped.
+func Tone(freq uint32) error {
+	ok, err := hasSound()
```
Comment (dlech) [inline speaker.go]: It seems a bit redundant to check this more than once.

File: speaker.go (hunk 2 above)
Comment (kortschak) [inline speaker.go]: This reflects the error that is used in `buttons.go`. Fixing the errors that currently use `fmt.Errorf` is part of #24.

File: ev3dev.go (hunk 1 above)
Comment (kortschak) [inline ev3dev.go]: Moving this out to "ev3go/ev3" and "ev3go/evb".

File: speaker.go (hunk 3 above)
Comment (kortschak) [inline speaker.go]: This and the path issue mean that a value that knows about the file and the path (and retains the buffer) is worth while. As a happy consequence, do *all* these gives exactly the same tone behaviour as the example you posted in the issue (previously there was an obvious lag that was not fixed by omitting this check).

**Comment by kortschak** [comment]
Tested on both platforms.

**Comment by kortschak** [comment]
*ping*

**Comment by dlech** [comment]
I didn't get a chance to actually test it out yet, but it looks good to me.

**Pull request merged**